src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -87,7 +102,7 @@ export function ThemeProvider({
   const style = {...customProperties, ...(!isParentThemeProvider && {color})};
 
   return (
-    <ThemeContext.Provider value={{...theme, textColor: color}}>
```

Comment:
Previously we stored a new value in ThemeContext on every render. This stops `parentContext` in the above useMemo dependencies  changing on every update, so now it's just `themeConfig` breaking the memoisation.